### PR TITLE
Migrate `getWeekOfMonth` to TypeScript

### PR DESCRIPTION
**Pull request by varwasabi**
I've addressed previous feedback here and have re-opened the PR. @kossnocorp please let me know if there's anything else for this. I also note that after running `./scripts/build/build.sh`, there's quite a lot of changes. I've opened this under the assumption that those files will be committed at a later time once the major version bump takes place.

### CODE REVIEW

File: src/getWeekOfMonth/index.ts [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+import { LocaleOptions, WeekStartOptions } from '../types'
 import startOfMonth from '../startOfMonth/index'
-import toInteger from '../_lib/toInteger/index'
-import requiredArgs from '../_lib/requiredArgs/index'
+
+// prettier-ignore
+type Options = {
+  locale?: LocaleOptions;
+  weekStartsOn?: WeekStartOptions;
+}
```
Comment (kossnocorp) [inline src/getWeekOfMonth/index.ts:12]:
```suggestion
```

File: src/getWeekOfMonth/index.ts [hunk 2]
```diff
@@ -29,44 +36,37 @@ import requiredArgs from '../_lib/requiredArgs/index'
  * var result = getWeekOfMonth(new Date(2017, 10, 9))
  * //=> 2
  */
-export default function getWeekOfMonth(date, dirtyOptions) {
+export default function getWeekOfMonth(
+  date: Date | number,
+  dirtyOptions: Options
```
Comment (kossnocorp) [inline src/getWeekOfMonth/index.ts:41]:
```suggestion
  dirtyOptions: LocaleOptions & WeekStartOptions
```

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Thanks, looks good besides the confusion with the options type.

File: src/getWeekOfMonth/index.ts [hunk 3]
```diff
@@ -29,44 +30,37 @@ import requiredArgs from '../_lib/requiredArgs/index'
  * var result = getWeekOfMonth(new Date(2017, 10, 9))
  * //=> 2
  */
-export default function getWeekOfMonth(date, dirtyOptions) {
+export default function getWeekOfMonth(
+  date: Date | number,
+  dirtyOptions: LocaleOptions & WeekStartOptions
```
Comment (kossnocorp) [inline src/getWeekOfMonth/index.ts:35]:
```suggestion
  dirtyOptions?: LocaleOptions & WeekStartOptions
```

**Comment by varwasabi** [comment]
@kossnocorp Thanks for the feedback and the assistance, should be ready to merge now.

File: src/getWeekOfMonth/index.ts [hunk 4]
```diff
@@ -29,44 +30,37 @@ import requiredArgs from '../_lib/requiredArgs/index'
...
-  var weekStartsOn =
+  const weekStartsOn =
     options.weekStartsOn == null
       ? defaultWeekStartsOn
       : toInteger(options.weekStartsOn)
 
-  // Test if weekStartsOn is between 0 and 6 _and_ is not NaN
-  if (!(weekStartsOn >= 0 && weekStartsOn <= 6)) {
-    throw new RangeError('weekStartsOn must be between 0 and 6 inclusively')
-  }
```
Comment (kossnocorp) [inline src/getWeekOfMonth/index.ts:49]: This is a breaking change we plan to introduce in v3, but we still need this check while we migrate the library in v2. Please recover those lines.

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Please recover this exception and also migrate the test file.

**Pull request closed without merging**